src/python/pants/backend/shell/goals/test.py

```diff
@@ -23,12 +24,15 @@
 from pants.util.logging import LogLevel
 
 
+@dataclasses.dataclass(frozen=True)
 class TestShellCommandFieldSet(TestFieldSet):
     required_fields = (
         ShellCommandCommandField,
         ShellCommandTestDependenciesField,
     )
 
+    environment: EnvironmentField
```

Comment:
Maybe move this to a separate PR given `TestShellCommandFieldSet` is in a separate subsystem. Moreover, is `TestShellCommandFieldSet` missing any other fields which it should arguably have?

Reply:
This change was from https://github.com/pantsbuild/pants/pull/20319 merged here so the test worked.